Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -381,14 +386,18 @@ def latest_version(formula, args:)
           next
         end
 
-        # Do not preprocess the URL when livecheck.strategy is set to :page_match
-        url = if livecheck_strategy == :page_match
+        # Only preprocess the URL when it's appropriate
+        url = if STRATEGY_SYMBOLS_TO_SKIP_PREPROCESS_URL.include?(livecheck_strategy)
```

Comment:
My thinking is that `/releases/latest` URLs shouldn't ever be processed and keeping this guard in place ensures this continues to be true regardless of whether `#preprocess_url` is skipped for some strategies.

Removing this guard would mess up checks for any formulae that use a `/releases/latest` URL in a `livecheck` block (i.e., not updated to use `strategy :github_latest`). `PageMatch` checks only skip preprocessing when `strategy :page_match` is present in the `livecheck` block (i.e., checks that implicitly use `PageMatch` don't skip preprocessing). Homebrew/homebrew-core#66134 will address this for homebrew/core but other taps could be affected if we removed this.

There isn't any notable benefit to removing this simple guard but there are clear detriments under certain circumstances. With that in mind, I think we should simply keep this guard.

Reply:
Got it - thank you! Perhaps it would be helpful to add a comment to that block above explaining which cases it catches (formulae in non-homebrew/core taps + \<some conditions\> ) -- otherwise it's not obvious why we still check for `github.com`.

Reply:
Since I'm not the target audience, do you think something short like ```# Handles any `livecheck` blocks not using the `GithubLatest` strategy``` would be sufficient? This guard would also apply if a check that uses a `/releases/latest` URL mistakenly sneaks into homebrew/core, so I don't think we need to go into the details of when we're most likely to encounter this situation (e.g., non-core taps).